packages/gatsby/src/utils/__tests__/mett.ts

```diff
@@ -0,0 +1,303 @@
+import { mett } from "../mett"
+
+describe(`mett`, (): void => {
+  describe(`regular events`, (): void => {
+    it(`can be called multiple times`, (): void => {
+      expect((): unknown => mett()).not.toThrow()
+      expect((): unknown => mett()).not.toThrow()
+      expect((): unknown => mett()).not.toThrow()
+    })
+
+    it(`returns unique objects`, (): void => {
+      expect(mett() !== mett()).toBe(true)
+    })
+
+    it(`can register an event`, (): void => {
+      const met = mett()
+
+      expect((): void => met.on(`foo`, (): void => {})).not.toThrow()
+    })
+
+    it(`can remove an added event`, (): void => {
+      const met = mett()
+      const f = (): void => {}
+
+      expect((): void => met.on(`foo`, f)).not.toThrow()
+      expect((): void => met.off(`foo`, f)).not.toThrow()
+    })
+
+    it(`calls a registered event handler on emit`, (): void => {
+      const met = mett()
+      const spy = jest.fn()
+
+      met.on(`foo`, spy)
+      met.emit(`foo`, `pass`)
+
+      expect(spy).toHaveBeenCalledWith(`pass`, `foo`)
+    })
+
+    it(`does not call a removed event`, (): void => {
+      const met = mett()
+      const spy = jest.fn()
+
+      met.on(`foo`, spy)
+      met.off(`foo`, spy)
+      met.emit(`foo`, `pass`)
+
+      expect(spy).not.toHaveBeenCalledWith(`pass`, `foo`)
+    })
+
+    it(`can remove a handler from an event that did not have it`, (): void => {
+      const met = mett()
+      const f = (): void => {}
+
+      expect((): void => met.off(`foo`, f)).not.toThrow()
+    })
```

Comment:
I actually wish this would throw. Probably can't change this as it might cause unexpected problems. But I think it would make a more prudent API if it did throw.

Reply:
I agree, it's reasonably likely there's a bug if this happens, or ugly code otherwise. But also agree that it's a non-trivial change to confirm.